src/main/java/uk/gov/pay/products/resources/ProductResource.java

```diff
@@ -88,6 +105,38 @@ public Response findProductByGatewayAccountIdAndExternalId(@PathParam("gatewayAc
                         Response.status(NOT_FOUND).build());
     }
 
+    @POST
+    @Path("/products/{productExternalId}/regenerate-api-key")
+    @Produces(APPLICATION_JSON)
+    @Consumes(APPLICATION_JSON)
+    public Response regenerateProductApiKey(@PathParam("productExternalId") String productExternalId) {
+        return productFactory.productFinder().findByExternalId(productExternalId).map((product) -> {
+            CreateTokenRequest createTokenRequest = new CreateTokenRequest(
+                    String.valueOf(product.getGatewayAccountId()),
+                    getDescription(product),
+                    emailAddress,
+                    CARD,
+                    PRODUCTS);
+
+            Response response = client.target(publicAuthUrl)
```

Comment:
I think `publicAuthUrl` is going to need the path adding to it somewhere since it’s just the hostname now (unless that happens somewhere else and I’ve missed it).

Reply:
It is in the config file.